File: app/src/main/java/com/dev/moviedb/model/MovieTrailer.java
package com.dev.moviedb.model;

import android.os.Parcel;
import android.os.Parcelable;

import java.util.ArrayList;
import java.util.List;

/**
 * This class represents the trailers associated with a certain movie.
 */
public class MovieTrailer implements Parcelable {

    private static final String YOUTUBE_HOST = "https://www.youtube.com/watch?v=%s&amp;feature=youtube_gdata";

    private List<String> mTrailersLink;

    public MovieTrailer(){
        mTrailersLink = new ArrayList<>();
    }

    public void addTrailer(String path){
        String url = String.format(YOUTUBE_HOST, path);
        mTrailersLink.add(url);
    }

    public List<String> getTrailersLink() {
        return mTrailersLink;
    }

    /*
    "trailers": {
        "quicktime": [],
        "youtube": [
          {
            "name": "Ant-Man Official Trailer #1",
            "size": "HD",
            "source": "ZiS7akYy4yA",
            "type": "Trailer"
          }
        ]
    }
     */


    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeStringList(this.mTrailersLink);
    }

    protected MovieTrailer(Parcel in) {
        this.mTrailersLink = in.createStringArrayList();
    }

    public static final Parcelable.Creator<MovieTrailer> CREATOR = new Parcelable.Creator<MovieTrailer>() {
        public MovieTrailer createFromParcel(Parcel source) {
            return new MovieTrailer(source);
        }

        public MovieTrailer[] newArray(int size) {
            return new MovieTrailer[size];
        }
    };
}

File: app/src/main/java/com/dev/moviedb/mvvm/repository/local/db/entity/content/MovieTrailer.kt
package com.dev.moviedb.mvvm.repository.local.db.entity.content

import android.os.Parcel
import android.os.Parcelable

/**
 * This class represents the movie trailers associated with a certain movie.
 *
 * Yamda 1.0.0.
 */
class MovieTrailer():Parcelable {

    private val YOUTUBE_HOST = "https://www.youtube.com/watch?v=%s&amp;feature=youtube_gdata";

    private var trailersLink : MutableList<String> = ArrayList()

    fun addTrailer(path: String){
        trailersLink.add("$YOUTUBE_HOST$path")
    }

    fun getTrailers(): Iterable<String> = trailersLink.asIterable()

    constructor(parcel: Parcel) : this() {
        parcel.readStringList(trailersLink)
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeStringList(trailersLink)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<MovieTrailer> {
        override fun createFromParcel(parcel: Parcel): MovieTrailer {
            return MovieTrailer(parcel)
        }

        override fun newArray(size: Int): Array<MovieTrailer?> {
            return arrayOfNulls(size)
        }
    }
}